### DISCUSSION TITLE
Connecting Ibis to aws athena

### DISCUSSION THREAD

**Discussion by uramith** [ibis-project/ibis#7229] [General]
Hey all, just wondering, is there any way to make connection to aws athena from Ibis ? If yes, please point me towards right resources.

**Comment by lostmygithubaccount** [comment]
hello @uramith, I believe this is possible through the Postgres backend: https://ibis-project.org/backends/postgresql

though is not something that would be considered officially supported. if you run into issues with the Postgres backend, can you let us know?

**Reply by uramith** [reply]
Hi @lostmygithubaccount ,

Thanks for the reply and sorry for the delayed response here, I was bit occupied with other work so couldn't able spend time on this. 

I had a look at the postgresql backends, but wondering about making a connection to athena using postgresql. At the moment All I have is aws credentials like **AccessKeyId** and **SecretAccessKey**.  I am not sure how to pass these in the args. 

If possible, could you please post a sample code snippet to make a connection to aws athena using postgresql backend ?

**Reply by lostmygithubaccount** [reply]
apologies for the slow reply, created an issue for tracking: https://github.com/ibis-project/ibis/issues/7682

this is likely not something we'll get to urgently, but we are having discussions about separating out connection logic from compilation logic that would enable AWS Athena and other services more easily going forward

**Reply by uramith** [reply]
@lostmygithubaccount Thank you for considering this issue. I really appreciate it. Looking forward to AWS authentication features in Ibis!
